servers/rendering/renderer_scene_cull.cpp

```diff
@@ -2271,10 +2350,13 @@ void RendererSceneCull::_frustum_cull(FrustumCullData &cull_data, FrustumCullRes
 
 	RID instance_pair_buffer[MAX_INSTANCE_PAIRS];
 
+	Transform inv_cam_transform = cull_data.cam_transform.inverse();
+	float z_near = cull_data.camera_matrix->get_z_near();
+
 	for (uint64_t i = p_from; i < p_to; i++) {
 		bool mesh_visible = false;
 
-		if (cull_data.scenario->instance_aabbs[i].in_frustum(cull_data.cull->frustum)) {
+		if (cull_data.scenario->instance_aabbs[i].in_frustum(cull_data.cull->frustum) && !_occlusion_cull(cull_data, i, cull_data.cam_transform.origin, inv_cam_transform, z_near)) {
```

Comment:
Give this is called so much, I would do it inside the if() and only if a bool is enabled or something (that is set globally or in the cull data), else it will most likely have a performance hit.